src/converters/downcast.js

```diff
@@ -0,0 +1,492 @@
+
+			const insertPosition = ViewPosition.createAt( trElement, 'end' );
+
+			createViewTableCellElement( tableWalkerValue, tableAttributes, insertPosition, conversionApi, options );
+		}
+
+		const viewPosition = conversionApi.mapper.toViewPosition( data.range.start );
+
+		conversionApi.mapper.bindElements( table, asWidget ? tableWidget : tableElement );
+		conversionApi.writer.insert( viewPosition, asWidget ? tableWidget : tableElement );
+	}, { priority: 'normal' } );
+}
+
+/**
+ * Model row element to view <tr> element conversion helper.
+ *
+ * This conversion helper creates whole <tr> element with child elements.
+ *
+ * @returns {Function} Conversion helper.
+ */
+export function downcastInsertRow( options = {} ) {
+	return dispatcher => dispatcher.on( 'insert:tableRow', ( evt, data, conversionApi ) => {
+		const tableRow = data.item;
+
+		if ( !conversionApi.consumable.consume( tableRow, 'insert' ) ) {
+			return;
+		}
+
+		const table = tableRow.parent;
+
+		const tableElement = conversionApi.mapper.toViewElement( table );
+
+		const row = table.getChildIndex( tableRow );
+
+		const tableWalker = new TableWalker( table, { startRow: row, endRow: row } );
+
+		const tableAttributes = {
+			headingRows: parseInt( table.getAttribute( 'headingRows' ) || 0 ),
+			headingColumns: parseInt( table.getAttribute( 'headingColumns' ) || 0 )
+		};
+
+		for ( const tableWalkerValue of tableWalker ) {
+			const tableSection = getOrCreateTableSection( getSectionName( row, tableAttributes ), tableElement, conversionApi );
+			const trElement = getOrCreateTr( tableRow, row, tableSection, conversionApi );
+
+			// Consume table cell - it will be always consumed as we convert whole row at once.
+			conversionApi.consumable.consume( tableWalkerValue.cell, 'insert' );
+
+			const insertPosition = ViewPosition.createAt( trElement, 'end' );
+
+			createViewTableCellElement( tableWalkerValue, tableAttributes, insertPosition, conversionApi, options );
+		}
+	}, { priority: 'normal' } );
+}
+
+/**
+ * Model tableCEll element to view <td> or <th> element conversion helper.
+ *
+ * This conversion helper will create proper <th> elements for tableCells that are in heading section (heading row or column)
+ * and <td> otherwise.
+ *
+ * @returns {Function} Conversion helper.
+ */
+export function downcastInsertCell( options = {} ) {
+	return dispatcher => dispatcher.on( 'insert:tableCell', ( evt, data, conversionApi ) => {
+		const tableCell = data.item;
+
+		if ( !conversionApi.consumable.consume( tableCell, 'insert' ) ) {
+			return;
+		}
+
+		const tableRow = tableCell.parent;
+		const table = tableRow.parent;
+
+		const tableWalker = new TableWalker( table );
+
+		const tableAttributes = {
+			headingRows: parseInt( table.getAttribute( 'headingRows' ) || 0 ),
+			headingColumns: parseInt( table.getAttribute( 'headingColumns' ) || 0 )
+		};
+
+		// We need to iterate over a table in order to get proper row & column values from a walker
```

Comment:
Can't we start iterating from the proper row already (get the row index straight from the model: `data.item.parent.index`).

Reply:
Yep, adding `startRow` & `endRow` will do the trick but anyway the `tableWalkerValue` is needed to properly calculate if this table cell is in heading section so the `tableWalker` is still required.